packages/firestore/test/integration/browser/webchannel.test.ts

```diff
@@ -32,6 +32,54 @@ const describeFn =
       xdescribe;
 
 describeFn('WebChannel', () => {
+  it.only('receives error messages', done => {
+    const projectId = DEFAULT_PROJECT_ID;
+    const info = getDefaultDatabaseInfo();
+    const conn = new WebChannelConnection(info);
+    const stream = conn.openStream<api.ListenRequest, api.ListenResponse>(
+      'Listen',
+      null
+    );
+
+    // Test data
+    const payload = {
+      database: 'projects/' + projectId + '/databases/(default)',
+      addTarget: {
+        query: {
+          parent: 'projects/' + projectId + '/databases/(default)',
+          structuredQuery: {
+            from: [{ collectionId: 'WHAT_COLLECTION_ID_SHOULD_THIS_USE' }]
```

Comment:
@schmidt-sebastian - The stats event doesn't get raised if there's isn't a successful message after open. I need a payload that will succeed. Can you recommend anything?

Reply:
The correct response to this message is https://source.corp.google.com/piper///depot/google3/google/firestore/v1beta1/firestore.proto;l=932?q=firestore.proto

This test should be mocking the WebChannel connection though, so I am not convinced that we need any actual message here. You should be able to invoke the callbacks in the right order even without sending a message.

Reply:
> The correct response to this message is...

I'm not sure what to do w/ the proto. If there's an error in web channel the stats event under time to first byte never gets raised. I either need to pass in the correct collection id (which I don't know how to get) or make a request that is guaranteed to succeed. If I'm not being clear perhaps we can clear this up over GVC?

> This test should be mocking the WebChannel connection ...

Are you sure about that? The file is listed as a browser integration test and when I turn on debug logging I'm seeing network traffic. If this is supposed to be mocked I can't tell where.

Reply:
Every time I look at this test I think it is a unit tests. But it turns out I am wrong every time. 

You should be able to use any collection ID - they don't have to exist to be queried. Do you get any responses? My bet would be that the project ID is wrong.

Reply:
I pushed my code with the default project id and settings and it's failing w/ an internal assertion error on CI. Not sure if this is related or not.

Reply:
Here is a patch: https://gist.github.com/schmidt-sebastian/302d74c42d81117eb129fa34731caddf (includes the diff above).

The main issues were:
- The 'detectBufferingProxy' flag wasn't set
- The onMessage/onStat event were racing. If onMessage was executed before, it would invoke `close()` which suppressed the stat event.

With those changes, I can get the test to pass.

Reply:
Thanks!